system/lib/wasmfs/file.h

```diff
@@ -63,6 +63,14 @@ class File : public std::enable_shared_from_this<File> {
     time_t& mtime() { return file->mtime; }
     time_t& atime() { return file->atime; }
 
+    ino_t getIno() {
+      // Set inode number to the file pointer. This gives a unique inode number.
+      // TODO: For security it would be better to use an indirect mapping.
+      // Ensure that the pointer will not overflow an ino_t.
+      static_assert(sizeof(file.get()) <= sizeof(ino_t));
+      return (ino_t)file.get();
+    }
```

Comment:
This can probably be a method directly on `File` rather than `File::Handle`, since the inode and underlying address are immutable. Taking the lock does not have any benefit here.

Reply:
I see, locking doesn't have any benefit here it seems.